### PR TITLE
py3 only

### PR DESCRIPTION
**Pull request by will-moore**
.

### CODE REVIEW

**Comment by joshmoore** [comment]
Probably doesn't hurt, but I think this is the one repo that might not _need_ the restriction.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by joshmoore** [comment]
```
./omero_marshal/legacy/affinetransform.py:29:21: E999 SyntaxError: invalid syntax
...etc...
```

**Comment by will-moore** [comment]
Failing with:
```
Processing dependencies for omero-marshal==0.6.4.dev1
Searching for omero-marshal==0.6.4.dev1
Reading https://pypi.org/simple/omero-marshal/
No local packages or working download links found for omero-marshal==0.6.4.dev1
error: Could not find suitable distribution for Requirement.parse('omero-marshal==0.6.4.dev1')
```

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#107](https://idr1-slot3.openmicroscopy.org:48443/jenkins/job/OMERO-python-superbuild-push/107/). See the [console output](https://idr1-slot3.openmicroscopy.org:48443/jenkins/job/OMERO-python-superbuild-push/107/consoleText) for more details.
Possible conflicts:
  - PR #29 chris-allan 'Initial support for FileAnnotation and OriginalFile'
    - tests/unit/test_shape_decoders.py

--conflicts

**Comment by will-moore** [comment]
Travis now failing with:
```
$ sudo python3 setup.py sdist install --record files.txt
Traceback (most recent call last):
  File "setup.py", line 4, in <module>
    from setuptools import setup, find_packages
ImportError: No module named 'setuptools'
```

I thought setuptools is built-in with python3, so not sure why this is failing. cc @manics @joshmoore @chris-allan

**Comment by chris-allan** [comment]
Pretty sure this is not happening during the regulary build because first you're in a virtualenv with all the python3 (3.6) state set up for you. You'll have to play around a bit on Travis. Ubuntu 16.04, which is what your build is defaulting to, has potential issues with setuptools compatibility and Python 3.5 is the actual default there.

I'm not aware of any differences as far as setuptools availability is concerned between python2 vs python3.

File: .travis.yml [hunk 1]
```diff
@@ -20,24 +20,26 @@ before_install:
...
+  - pip3 install flake8
+  - pip3 install -U setuptools
   - flake8 .
 
 script:
-    - PYTHONPATH=/tmp/OMERO.py/lib/python python setup.py test --pytest-args='-v -v'
-    - sudo python setup.py sdist install --record files.txt
-    - PYTHONPATH=/tmp/OMERO.py/lib/python python -c 'import omero.clients; import omero_marshal.encode'
+    - python3 setup.py test --pytest-args='-v -v'
+    - sudo python3 setup.py sdist install --record files.txt
```
Comment (manics) [inline .travis.yml]: Mixing sudo and non-sudo  python/pip commands may mean you're using different python environments depending on how the PATH is setup.

Comment (will-moore) [reply .travis.yml]: OK, I'll remove usage of sudo...

**Comment by joshmoore** [comment]
Rolling into 0.7.0.dev1 as the first version without support for Python 2.

**Pull request merged**